js/modules/k6/execution/execution.go

```diff
@@ -362,6 +379,61 @@ func (o *tagsDynamicObject) Keys() []string {
 	for k := range tagsMap {
 		keys = append(keys, k)
 	}
+	return keys
+}
+
+type metadataDynamicObject struct {
+	runtime *goja.Runtime
+	state   *lib.State
+}
+
+// Get a property value for the key. May return nil if the property does not exist.
+func (o *metadataDynamicObject) Get(key string) goja.Value {
+	tcv := o.state.Tags.GetCurrentValues()
+	if metadatum, ok := tcv.Metadata[key]; ok {
+		return o.runtime.ToValue(metadatum)
+	}
+	return nil
+}
+
+// Set a property value for the key. It returns true if succeed. String, Boolean
+// and Number types are implicitly converted to the goja's relative string
+// representation. An exception is raised in case a denied type is provided.
+func (o *metadataDynamicObject) Set(key string, val goja.Value) bool {
+	var err error
+	o.state.Tags.Modify(func(tagsAndMeta *metrics.TagsAndMeta) {
+		err = common.ApplyCustomUserMetadata(tagsAndMeta, key, val)
+		if err != nil {
+			panic(o.runtime.NewTypeError(err.Error()))
+		}
+	})
+	return true
+}
```

Comment:
Is there a reason the `err` is declared outside of the closure?

I suppose this method used to return it before, and now we always return `true`, but it can just be scoped within the closure. And the `err != nil` check can be done on the same line.

Same goes for `tagsDynamicObject.Set()`, since this was copied from there, so we might as well update it there too.

Reply:
I guess that is true - I just copied and edited the code from the tags. I did try to figure out if I can make it more generic - but unfortunately that seems like *way* more work then it would save code.

Reply:
fixed in [f011c23](https://github.com/grafana/k6/pull/3037/commits/f011c23571315c37102a709ad00cda42a9a6ecc9)